Added a lot of integration for Francois

jeffaburke opened the pull request.

Rohan-Sreenivasan committed:
Added Indexing Map

Rohan-Sreenivasan committed:
THE ROBOT LIVES

FRANCOIS IS ALIVE

Rohan-Sreenivasan committed:
IR values changed

jeffaburke committed:
Fixed Lift Brake

Rohan-Sreenivasan committed:
Averaging Drive Code

The drive code uses modifiers to take a rolling average of all the drive inputs. The Current code will take 10 samples for  given amount of time

Rohan-Sreenivasan committed:
Merge remote-tracking branch 'origin/Integration' into Integration

jeffaburke committed:
Added End Game Lift

jeffaburke committed:
Merge remote-tracking branch 'origin/Integration' into Integration

Rohan-Sreenivasan committed:
Shoot all command

Command with 2 seq command groups to shoot all balls in the system

jeffaburke committed:
Changed Drive and Lift

DRIVE IS BROKEN

jeffaburke committed:
Merge remote-tracking branch 'origin/Integration' into Integration

jeffaburke committed:
Fixed to Build

jeffaburke committed:
FIxed Drive

jeffaburke committed:
Added Trigger Package

jeffaburke committed:
Working lift

msoucy commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -86,7 +91,7 @@ public void robotInit() {
 
         drive.setDefaultCommand(drive.drive(driveController::getTriggers, () -> driveController.getX(Hand.kLeft)));
-        lift.setDefaultCommand(lift.moveLift(copilotController::getTriggers));
+        lift.setDefaultCommand(lift.moveLift(() -> driveController.getY(Hand.kRight)));
```
I expect that the wiki page on controls will be updated?

jeffaburke replied:
Added it

msoucy commented on src/main/java/frc/robot/maps/FrancoisMap.java:
```diff
@@ -55,45 +61,49 @@ public EncodedSpeedController getLeft() {
-                };
+                return EncodedSpeedController.join(averageInputs, enc);
             }
+
```
Eew, ugly comments. 🙅‍♂

bot190 replied:
I think I cleaned up whatever this referred to.

msoucy commented on src/main/java/frc/robot/subsystems/Drive.java:
```diff
@@ -4,6 +4,7 @@
 
 import com.chopshop166.chopshoplib.maps.DifferentialDriveMap;
 import com.chopshop166.chopshoplib.outputs.EncodedSpeedController;
+import com.chopshop166.chopshoplib.outputs.ModSpeedController;
```
Did you actually use this import?

msoucy commented on src/main/java/frc/robot/subsystems/Drive.java:
```diff
@@ -56,6 +57,7 @@ public Drive(DifferentialDriveMap map) {
      * @return returns a run command so drive will stay running as long as drive is
      *         being called
      */
+
```
😞

bot190 replied:
(Matt Translator): Blank line here should be removed

msoucy replied:
👍

jeffaburke replied:
Fixed

msoucy commented on src/main/java/frc/robot/subsystems/Indexer.java:
```diff
@@ -155,6 +154,26 @@ public CommandBase unLoadBall() {
+
+        }, () -> {
+
+            shootingBalls();
```
This will make a command (group), and then... do nothing with it.

msoucy commented on src/main/java/frc/robot/subsystems/Lift.java:
```diff
@@ -1,16 +1,20 @@
 package frc.robot.subsystems;
 
+import java.sql.Time;
```
🚫

msoucy commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -165,21 +170,37 @@ public SequentialCommandGroup passLine() {
+        driveController.getButton(Button.kB).whenPressed(indexer.shootingBalls());
+        // driveController.getButton(Button.kX).whenPressed(endGame());
+        XboxTrigger endTrigger = new XboxTrigger(driveController, 5, false);
+        // endTrigger.and(new EndGameTrigger(120)).whenActive(endGame());
```
Why is this commented out?

bot190 replied:
Stuff wasn't working and we think this was why. We need to debug further at some point.

msoucy commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -165,21 +170,37 @@ public SequentialCommandGroup passLine() {
-        copilotController.getButton(Button.kX).whenHeld(cancelCommand());
-        driveController.getButton(Button.kB).whenPressed(indexer.unLoadBall());
+        driveController.getButton(Button.kB).whenPressed(indexer.shootingBalls());
+        // driveController.getButton(Button.kX).whenPressed(endGame());
```
Why is this commented out?

bot190 replied:
I agree, this can be removed entirely.

msoucy commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -68,13 +70,16 @@
-        nameEntry.setPersistent();
-        nameEntry.setDefaultString("Unknown");
-        Shuffleboard.getTab("RobotData").addString("RobotName", () -> nameEntry.getString("Unknown"));
+        // nameEntry.setPersistent();
```
What happened here?

jeffaburke replied:
None of that is needed

jeffaburke committed:
Minor Fixes by Ben

bot190 committed:
Merge remote-tracking branch 'origin/master' into Integration

bot190 committed:
Some cleanup in Robot.java

bot190 committed:
Add names to command groups

bot190 committed:
Add safety to lift ratchet release

The lift ratchets require a careful sequence to release before we can drive the elevator up. If the elevator is not stowed correctly we may not be able to release the ratchets. To prevent us from stalling the motors for a prolonged time we will timeout the initial move (to release tension) and will cancel the rest of the command group.

bot190 commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -165,18 +166,37 @@ public void testInit() {
 
-    public SequentialCommandGroup passLine() {
-        return new SequentialCommandGroup(drive.driveDistance(40, .5));
+    // will spin the shooter then shoot all the balls and then turn the shooter off.
```
This needs to be moved ABOVE `shootAllBalls()`

bot190 commented on src/main/java/frc/robot/maps/FrancoisMap.java:
```diff
@@ -19,15 +21,17 @@
     @Override
     public DifferentialDriveMap getDriveMap() {
-        final double distancePerPulse = (1.0 / 46.0) * (1.0 / 12.27) * (6.0 * Math.PI);
+        final double distancePerPulse = (1.0 / 12.27) * (6.0 * Math.PI);
```
This should have a comment explaining what the numbers mean.

msoucy replied:
The numbers, Mason. What do they mean?

jeffaburke replied:
Fixed

bot190 commented on src/main/java/frc/robot/maps/FrancoisMap.java:
```diff
@@ -98,8 +103,51 @@ public ControlPanelMap getControlPanelMap() {
             }
+
+            public BooleanSupplier topPierreIR() {
+                AnalogTrigger topPierreIR = new AnalogTrigger(0);
```
We should move these out of the functions (similar to drive) to appease the java gods.

jeffaburke replied:
Gods Appeased

bot190 commented on src/main/java/frc/robot/subsystems/Indexer.java:
```diff
@@ -86,103 +91,91 @@ public CommandBase reverseOutput() {
                 singulator.set(singulatorMotorSpeed);
             }
-            if (backIntakeIR.getAsBoolean()) {
+            if (backIntakeIR.getAsBoolean() && !topPierreIR.getAsBoolean()) {
```
We should add comments explaining why we did this since it's non-obvious.
Also reading the docs `isFinished` gets called IMMEDIATELY after `execute` so I have no idea why the motors have time to run.

jeffaburke replied:
FIxed

bot190 commented on src/main/java/frc/robot/subsystems/Lift.java:
```diff
@@ -53,13 +57,15 @@ public Lift(RobotMap.LiftMap map) {
             }
-        } else if (speed <= 0) {
-            elevatorBrake.set(true);
+        } else if (speed > 0) {
```
We need to be careful with the `>` and `<` here as they will likely change when we move to the triggers. The motion convention should be made clear in a comment at the top. (Positive input lowers the lift, negative raises it)

jeffaburke replied:
Outdated We fixed this in Robot using the inverse of the Trigger values

bot190 commented on src/main/java/frc/robot/subsystems/Shooter.java:
```diff
@@ -50,7 +50,8 @@ public void periodic() {
 
     public CommandBase spinUp() {
```
Do we need to pull in some of @AustinFortunato's shooter code?

jeffaburke replied:
Probably but when he re-bases with master it should

bot190 reviewed (changes requested):
Lots of stuff here. Lets make sure we close these out quickly so we can keep making progress.

jeffaburke committed:
More Integration Stuff (Named By Ben)

jeffaburke committed:
Fixed Index and shoot all balls

msoucy commented on CommandButtonMappings.md:
```diff
@@ -6,7 +6,8 @@ Lift | Elevator Up | Copilot Controller: Right Trigger
-Shooter | Stop Motor | Drive Controller: B
-Drive | Drive | Drive Controller: Arcade Drive (Triggers for speed; left joystick    for turning)
+Shooter | Spin Motor | Drive Controller: Right Bumper
+Shooter | Stop Motor | Drive Controller: Stop Motor
```
What button is this?

jeffaburke replied:
Disregard this file I'll rebase

msoucy commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -75,13 +78,11 @@
     @Override
     public void robotInit() {
         configureButtonBindings();
-        nameEntry.setPersistent();
```
Why's this gone?

jeffaburke replied:
Added Back

msoucy commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -165,18 +166,37 @@ public void testInit() {
 
-    public ParallelCommandGroup cancelCommand() {
+    public CommandBase passLine() {
+        CommandBase cmd = new SequentialCommandGroup(drive.driveDistance(40, .5));
```
Why is this a group?

jeffaburke replied:
I will be adding a shooting function to it soon once Austin finishes with his code

msoucy replied:
👍

msoucy commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -186,14 +206,29 @@ public SequentialCommandGroup passLine() {
+        XboxTrigger endTrigger = new XboxTrigger(copilotController, Hand.kRight);
+        endTrigger.and(new EndGameTrigger(120)).whenActive(endGame());
+
+        // driveController.getButton(Button.kBumperRight).whenPressed(shooter.spinUp());
```
![](https://imgs.xkcd.com/comics/commented.png)

jeffaburke replied:
FIxed Lol

msoucy commented on src/main/java/frc/robot/subsystems/Drive.java:
```diff
@@ -76,10 +84,12 @@ public CommandBase driveDistance(double distance, double speed) {
             double avg = (leftMotorGroup.getEncoder().getDistance() + rightMotorGroup.getEncoder().getDistance()) / 2;
             return (avg >= distance);
         }, this);
+        cmd.setName("Drive Distane");
```
Spelling is hard

jeffaburke replied:
Yikes Ben Can't Spell at 12 at night

jeffaburke replied:
If your Ben at Midnight then yes, yes it is

msoucy commented on src/main/java/frc/robot/subsystems/Indexer.java:
```diff
@@ -86,103 +91,85 @@ public CommandBase reverseOutput() {
     // this will bring the ball to the shooter, it must already be at the top
 
+    public CommandBase shootAllBalls() {
+        CommandBase cmd = new SequentialCommandGroup(shootingBalls(), shootingBalls(), shootingBalls(), shootingBalls(),
```
I think `CommandUtils.repeat(5, this::shootingBalls);` is the line you are looking for.

jeffaburke replied:
ok will try it

jeffaburke replied:
Type mismatch: cannot convert from Command to CommandBase Java(16777233)

jeffaburke replied:
Fixed

msoucy commented on src/main/java/frc/robot/subsystems/Indexer.java:
```diff
@@ -49,22 +48,28 @@ public Indexer(final IndexMap map) {
 
-    public SequentialCommandGroup shootOneBall() {
-        return new SequentialCommandGroup(loadBallToTop(), unLoadBall());
+    public CommandBase shootingBalls() {
```
Why is it called `shootingBalls`? why the `ing`? Why plural balls?

jeffaburke replied:
Don't Know *cough* *cough* @Rohan-Sreenivasan

jeffaburke replied:
*Fixed

msoucy commented on src/main/java/frc/robot/subsystems/Lift.java:
```diff
@@ -83,14 +89,51 @@ private liftHeights(double mPosition) {
+                // This should get the command group that is trying to move the elevator
+                // We will then cancel that command group so we stop.
+                // TODO Send something to the dashboard.
+                CommandScheduler.getInstance().requiring(this).cancel();
```
Did you test this?

jeffaburke replied:
Nope

bot190 replied:
It seems to work, though we didn't explicitly check it.

msoucy replied:
Good enough for me.

msoucy commented:
Merge conflicts 👏👏👏👏👏

andrewmartin5 committed:
Readded lower lift limit switch

jeffaburke committed:
Merge remote-tracking branch 'origin/master' into Integration

jeffaburke committed:
Merge remote-tracking branch 'origin/Integration' into Integration

msoucy commented on src/main/java/frc/robot/maps/FrancoisMap.java:
```diff
@@ -180,6 +180,11 @@ public BooleanSupplier upperLiftLimit() {
 
+            @Override
+            public BooleanSupplier lowerLiftLimit() {
+                return upperLimit::get;
```
This seems wrong.

jeffaburke replied:
Fixed

jeffaburke committed:
Fixed For Comments

jeffaburke committed:
Fixed for comments again

jeffaburke committed:
Fix 3 for comments

jeffaburke committed:
Auto stash before merge of "Integration" and "origin/master"

msoucy commented on src/main/java/frc/robot/subsystems/Drive.java:
```diff
@@ -46,6 +48,7 @@ public Drive(DifferentialDriveMap map) {
         leftMotorGroup = map.getLeft();
         gyro = map.getGyro();
         driveTrain = new DifferentialDrive(leftMotorGroup, rightMotorGroup);
+        leftMotorGroup.getEncoder();
```
Why?

jeffaburke replied:
### Fixed

jeffaburke committed:
Fixed for comments 4

msoucy reviewed (dismissed):
Good enough for now.

bot190 commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -165,18 +167,37 @@ public void testInit() {
 
-        return new ParallelCommandGroup(intake.discharge(), indexer.reversePush());
+    // will spin the shooter then shoot all the balls and then turn the shooter off.
+    public CommandBase shootAllBalls() {
```
Please add a `TODO` comment here saying that this doesn't work.

jeffaburke replied:
Done

bot190 commented on src/main/java/frc/robot/maps/FrancoisMap.java:
```diff
@@ -98,42 +103,93 @@ public ControlPanelMap getControlPanelMap() {
 
+            @Override
+            public BooleanSupplier lowerLiftLimit() {
+                return lowerLimit::get;
```
This is going to need to be inverted as well.

jeffaburke replied:
Fixed

bot190 commented on src/main/java/frc/robot/maps/FrancoisMap.java:
```diff
@@ -98,42 +103,93 @@ public ControlPanelMap getControlPanelMap() {
             public IEncoder getLiftEncoder() {
                 SparkMaxEncoder getEncoder = new SparkMaxEncoder(leader.getEncoder());
-
+                getEncoder.setScaleFactor(1);
```
Lets figure out the actual scale factor here.

jeffaburke replied:
Fixed

bot190 commented on src/main/java/frc/robot/subsystems/Indexer.java:
```diff
@@ -86,103 +91,86 @@ public CommandBase reverseOutput() {
     }
     // this will bring the ball to the shooter, it must already be at the top
 
+    public CommandBase shootAllBalls() {
```
We should figure out how to make this a `RepeatWhileCommand()`

bot190 commented on src/main/java/frc/robot/subsystems/Lift.java:
```diff
@@ -49,17 +54,20 @@ public Lift(RobotMap.LiftMap map) {
+            speed = 0.0;
+        }
+        if (speed < 0) {
+            if (upperLimitSwitch.getAsBoolean() || lowerLimitSwitch.getAsBoolean() || !elevatorBrake.get()) {
```
Only one limit switch should be here. The `lowerLimitSwitch` should be checked in the below `if (speed > 0)`.

bot190 commented on src/main/java/frc/robot/subsystems/Shooter.java:
```diff
@@ -50,7 +50,8 @@ public void periodic() {
     public CommandBase spinUp() {
         return new InstantCommand(() -> {
-            shooterWheelMotor.set(calculateRPM() / MAX_RPM);
+            // shooterWheelMotor.set(calculateRPM() / MAX_RPM);
```
Please add a `TODO` comment here to note that we need to incorporate the automatic calculations again.

jeffaburke replied:
Done

jeffaburke committed:
Merge remote-tracking branch 'origin/master' into Integration

jeffaburke committed:
Added New Chop Shop Lib Release plus comment fixes

bot190 commented on src/main/java/frc/robot/maps/FrancoisMap.java:
```diff
@@ -108,33 +156,43 @@ public LiftMap getLiftMap() {
+            PIDSparkMax pidLeader = new PIDSparkMax(leader);
             InvertDigitalInput upperLimit = new InvertDigitalInput(0);
+            InvertDigitalInput lowerLimit = new InvertDigitalInput(1);
+            double distancePerRev = (1.0 / 81.0) * (1.87 * Math.PI);
```
Diameter is 2.551" not 1.87, sorry.

jeffaburke replied:
Fixed

bot190 commented on src/main/java/frc/robot/subsystems/Lift.java:
```diff
@@ -83,14 +94,51 @@ private liftHeights(double mPosition) {
     }
 
+    public CommandBase disengageRatchet(DoubleSupplier speed) {
+        CommandBase cmd = new SequentialCommandGroup(moveTicks(.5, 0.10), turnOffBrake(), new WaitCommand(.1),
```
Make this 0.1" instead of 0.5"

jeffaburke replied:
FIxed

jeffaburke committed:
Merge branch 'master' into Integration

jeffaburke committed:
Fixes for Values

jeffaburke committed:
Merge Changes and Unused Import Fix
